src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/Envelope.java

```diff
@@ -107,4 +109,15 @@ public String getJurisdiction() {
     public String getZipFileName() {
         return zipFileName;
     }
+
+    public void setDocumentUrls(Map<String, String> response) {
+        scannableItems.forEach(
+            scannableItem -> Optional.ofNullable(response.computeIfAbsent(scannableItem.getFileName(), s -> null))
+                .ifPresent(scannableItem::setDocumentUrl)
+        );
+    }
```

Comment:
There are few ways to do this. To begin with, there's nothing wrong with getting old school and do a simple
```java
if (response.containsKey(key)) {
    scannableItem.setDocumentUrl(response.get(key));
}
```
Then if you insist there's slightly simple set of functional calls
```java
Optional.ofNullable(response.getOrDefault(key, null)).ifPresent(...)
```
There's nothing to compute in your case, but personally I would choose one of these samples above. I don't mind any approach